sklearn/ensemble/_stacking.py

```diff
@@ -0,0 +1,654 @@
+from ..linear_model import LinearRegression
+
+from ..model_selection import cross_val_predict
+from ..model_selection import check_cv
+
+from ..preprocessing import LabelEncoder
+
+from ..utils import Bunch
+from ..utils.metaestimators import _BaseComposition
+from ..utils.metaestimators import if_delegate_has_method
+from ..utils.multiclass import check_classification_targets
+from ..utils.validation import check_is_fitted
+from ..utils.validation import column_or_1d
+
+
+class _BaseStacking(_BaseComposition, MetaEstimatorMixin, TransformerMixin,
+                    metaclass=ABCMeta):
+    """Base class for stacking method.
+
+    Warning: This class should not be used directly. Use derived classes
+    instead.
+    """
+    _required_parameters = ['estimators']
+
+    @abstractmethod
+    def __init__(self, estimators, final_estimator=None, cv=None,
+                 stack_method='auto', n_jobs=None, verbose=0):
+        self.estimators = estimators
+        self.final_estimator = final_estimator
+        self.cv = cv
+        self.stack_method = stack_method
+        self.n_jobs = n_jobs
+        self.verbose = verbose
+
+    def _validate_final_estimator(self, default):
+        if self.final_estimator is not None:
+            self.final_estimator_ = clone(self.final_estimator)
+        else:
+            self.final_estimator_ = clone(default)
+
+    @staticmethod
+    def _method_name(name, estimator, method):
+        if estimator in (None, 'drop'):
+            return None
+        if method == 'auto':
+            if getattr(estimator, 'predict_proba', None):
+                return 'predict_proba'
+            elif getattr(estimator, 'decision_function', None):
+                return 'decision_function'
+            else:
+                return 'predict'
+        else:
+            if not hasattr(estimator, method):
+                raise ValueError('Underlying estimator {} does not implement '
+                                 'the method {}.'.format(name, method))
+            return method
+
+    def set_params(self, **params):
+        """Setting the parameters for the stacking estimator.
+
+        Valid parameter keys can be listed with `get_params()`.
+
+        Parameters
+        ----------
+        params : keyword arguments
+            Specific parameters using e.g.
+            `set_params(parameter_name=new_value)`. In addition, to setting the
+            parameters of the stacking estimator, the individual estimator of
+            the stacking estimators can also be set, or can be removed by
+            setting them to 'drop'.
+
+        Examples
+        --------
+        # In this example, the RandomForestClassifier is removed
+        clf1 = LogisticRegression()
+        clf2 = RandomForestClassifier()
+        eclf = StackingClassifier(estimators=[('lr', clf1), ('rf', clf2)]
+        eclf.set_params(rf='drop')
+        """
+        super()._set_params('estimators', **params)
```

Comment:
that looks like we can't set parameters for the final estimator? is that true/intended?

Reply:
No you can `est.set_params(final_estimators__whatever_param=xxxx)`
This is just allowing to access directly the estimator embedded in the list of `estimators`.